### ISSUE TITLE
Failed workflow during dispatcher development

### ISSUE THREAD

**Issue by burnout87** [oda-hub/dispatcher-plugin-gw#16]
During the development of https://github.com/oda-hub/dispatcher-app/pull/585 , the following workflow failed:

https://github.com/oda-hub/dispatcher-app/actions/runs/6149352313/job/16685038908

Please feel free to move the issue somewhere else more suitable if needed

**Comment by volodymyrss** [comment]
@dsavchenko could you have a look why it failed?

**Comment by dsavchenko** [comment]
I tried to reproduce locally, created the fresh environment with the corresponding branch of the dispatcher and all tests pass. Will dig the logs

**Comment by dsavchenko** [comment]
That's really weird. FileNotFoundError appears somewhere when calling `download_products` endpoint after successfull request for the products itself. That's all I can see from the logs.

@volodymyrss do you know if it's possible to fully reproduce the test environment used in github? Probably call pip freeze to produce an artifact or something?

**Comment by volodymyrss** [comment]
I do not know of an easy way to reproduce github action environment. But yes, package list can be tracked.

However, the problem seems to be intermittent. I suspect a race between writing the file and reading it. It's unclear why this happens on with nb services, could be because of size of the request.

Maybe we can ensure the file is flushed before returing from run_analysis?

**Comment by dsavchenko** [comment]
> However, the problem seems to be intermittent. I suspect a race between writing the file and reading it. It's unclear why this happens on with nb services, could be because of size of the request.

Hm, could be the case with large files. I'll try to check

**Comment by dsavchenko** [comment]
@burnout87  I see now that all tests in https://github.com/oda-hub/dispatcher-app/pull/585 passed
So is this issue gone?

**Comment by burnout87** [comment]
Yes, I re-run them and they all passed, but this happened few days during the development. I reported to make sure that I did not introduce any change that could potentially break something somewhere else.

But it seems like it is not the case

**Comment by burnout87** [comment]
For the same reason I opened https://github.com/oda-hub/dispatcher-plugin-nb2workflow/issues/72 too
